File: php/lib/FizzBuzzTest.php
<?php

require_once('./lib/FizzBuzz.php');

class FizzBuzzTest extends PHPUnit_Framework_TestCase {
    function testFizzBuzz() {
        $fizzbuzz = new FizzBuzz();
        $this -> assertEquals('1', $fizzbuzz->do(1));
        $this -> assertEquals('2', $fizzbuzz->do(2));
        $this -> assertEquals('Fizz', $fizzbuzz->do(3));
        $this -> assertEquals('Buzz', $fizzbuzz->do(5));
        $this -> assertEquals('FizzBuzz', $fizzbuzz->do(15));
    }
}

File: golang/fizzbuzz_test.go
package fizzbuzz

import (
	"testing"
)

func TestFizzBuzz(t *testing.T) {
	fb := NewFizzBuzz()
	var expected, result string
	expected = "1"
	result = fb.Do(1)
	if expected != result {
		t.Errorf("Expected:%s, but result is:%v", expected, result)
	}
	expected = "2"
	result = fb.Do(2)
	if expected != result {
		t.Errorf("Expected:%s, but result is:%v", expected, result)
	}
	expected = "Fizz"
	result = fb.Do(3)
	if expected != result {
		t.Errorf("Expected:%s, but result is:%v", expected, result)
	}
	expected = "Buzz"
	result = fb.Do(5)
	if expected != result {
		t.Errorf("Expected:%s, but result is:%v", expected, result)
	}
	expected = "FizzBuzz"
	result = fb.Do(15)
	if expected != result {
		t.Errorf("Expected:%s, but result is:%v", expected, result)
	}
}
